tests/test_pre.py

```diff
@@ -109,5 +110,110 @@ def test_correct_multiimage_illum(image_files_noise):
         assert np.median(k) > 100
 
 
+cellomics_pattern = "MFGTMP_150406100001_A01f{0:02d}d0.TIF"
+
+missing_test_fns = [
+    ([cellomics_pattern.format(i) for i in range(25)], []),
+    ([cellomics_pattern.format(i) for i in range(25)], [1, 13])
+]
+
+# delete "images" with fields 1 and 13 from second set of
+# image filesnames
+missing_test_fns[1][0].remove(cellomics_pattern.format(1))
+missing_test_fns[1][0].remove(cellomics_pattern.format(13))
+
+
+@pytest.mark.parametrize("fns, expected", missing_test_fns)
+def test_find_missing_fields(fns, expected):
+    actual = pre.find_missing_fields(fns)
+    np.testing.assert_array_equal(actual, expected)
+
+
+missing_mask_test = [
+    ([], [[0, 1, 2]], 10, 5, np.ones((10, 15), dtype=np.bool)),
+    ([0, 5], [[0, 1, 2], [4, 5, 6]], 5, 10, np.ones((10, 30), dtype=np.bool)),
+    ([3, 4], [[0, 1], [2, 3], [4, 5]], 10, 5, np.ones((30, 10), dtype=np.bool))
+]
+
+# insert False to missing areas of expected output
+missing_mask_test[1][4][0:5, 0:10] = False
+missing_mask_test[1][4][5:10, 10:20] = False
+
+missing_mask_test[2][4][10:20, 5:10] = False
+missing_mask_test[2][4][20:30, 0:5] = False
+
+
+@pytest.mark.parametrize("missing, order, rows, cols, expected",
+                         missing_mask_test)
+def test_create_missing_mask(missing, order, rows, cols, expected):
+    actual = pre.create_missing_mask(missing, order, rows, cols)
+    np.testing.assert_array_equal(actual, expected)
```

Comment:
Can you explain what this is doing? I've never seen this before! A comment above the usage would be good.